extensions/amp-story/1.0/amp-story.js

```diff
@@ -2237,13 +2240,32 @@ export class AmpStory extends AMP.BaseElement {
 
     const pagesByDistance = this.getPagesByDistance_();
 
-    this.mutateElement(() => {
+    const preloadAllPages = () => {
       pagesByDistance.forEach((pageIds, distance) => {
         pageIds.forEach((pageId) => {
           const page = this.getPageById(pageId);
           page.setDistance(distance);
         });
       });
+    };
+
+    this.mutateElement(() => {
+      if (
+        !isExperimentOn(this.win, 'amp-story-load-first-page-only') ||
+        !prioritizeActivePage
+      ) {
+        return preloadAllPages();
+      }
+
+      const activePageId = devAssert(pagesByDistance[0][0]);
+      new Promise((res, rej) => {
+        const page = this.getPageById(activePageId);
+        page.setDistance(0);
+        page.signals().whenSignal(CommonSignals.LOAD_END).then(res);
+        page.signals().whenSignal(EventType.SWITCH_PAGE).then(rej);
+      })
+        .then(preloadAllPages)
+        .catch();
```

Comment:
I know Ryan asked you to remove it but consistency is good, and this would lead to a non obvious bug if someone extracts this function into a method

```suggestion
        .then(() => preloadAllPages());
```